main.py

```diff
@@ -303,13 +309,178 @@ def extract_data(directory: str, has_dedicated_gpu: bool, gpu_in_cpu: bool, gui:
+
+    try:
+        with open(os.path.join(path, "copy_this_to_tarallo.json"), "x") as f:
+            f.write(output)
+    except FileExistsError:
+        with open(os.path.join(path, "copy_this_to_tarallo.json"), "w") as f:
+            f.write(output)
+
+    print(f"You can also transfer the generated JSON file $OUTPUT_PATH/copy_this_to_tarallo.json to your PC with 'scp {path}/copy_this_to_tarallo.json <user>@<your_PC's_IP>:/path/on/your/PC' right from this terminal.")
+
+
+def run_extract_data(path, args):
+    try:
+        data = extract_and_collect_data_from_generated_files(directory=path,
+                                                             has_dedicated_gpu=args.gpu,
+                                                             gpu_in_cpu=args.cpu,
+                                                             verbose=args.verbose,
+                                                             gui=False)
+        print_output(json.dumps(data, indent=2), path)
+    except InputFileNotFoundError as e:
+        print(str(e))
+        exit(1)
+
+def check_required_files(path):
+    file_in_dir = os.listdir(path)
+    with open("required_files.txt", "r") as f:
+        for line in f.readlines():
+            file = line.rstrip()
+
+            for ex in file_in_dir:
+                if fnmatch.fnmatch(ex, file):
+                    break
+            else:
+                print(f"[bold red]Missing file {file}\nPlease re-run this script without the -f or --files option.[/]")
+                exit(-1)
+
+
+
+def check_and_install_dependencies():
+    install_cmd = "apt install -y pciutils i2c-tools mesa-utils smartmontools dmidecode < /dev/null"
+    exit_value = os.system("dpkg -s pciutils i2c-tools mesa-utils smartmontools dmidecode > /dev/null")
+    if exit_value == 1:
+        ans = input("You need to install some packages in order for the peracotta to work. Do you want to install them? y/N ").lower()
+        if ans == 'y':
+            if os.geteuid() != 0:
+                os.system(f"sudo {install_cmd}")
+            else:
+                os.system(f"/bin/bash/ -c {install_cmd}")
+        else:
+            print("Quitting...")
+            exit(-1)
+
+
+def open_default_browser():
+    import base64
+    web_link = "aHR0cHM6Ly90YXJhbGxvLndlZWVvcGVuLml0L2J1bGsvYWRkCg=="
+    web_link = base64.b64decode(web_link).decode('ascii').rstrip()
+    egg = Console()
+    text = ['Congratulations!!!', "You're", "the", "1000th", "WEEEisitor", "of", "the", "day"]
+    this_moment = datetime.now()
+    if this_moment.minute == this_moment.second:
+        for word in text:
+            red = random.randint(0, 255)
+            green = random.randint(0, 255)
+            blue = random.randint(0, 255)
+            egg.print(word, end=" ", style=f"rgb({red},{green},{blue})")
+        egg.print(web_link)
+    else:
+        print(f"Finished successfully! Now you can add this output to T.A.R.A.L.L.O (bad timing, anyway) {web_link}")
```

Comment:
Let's not give any hints 🥸
```suggestion
        print(f"Finished successfully! Now you can add this output to T.A.R.A.L.L.O. {web_link}")
```